cephfs: addressed few todo

riya-singhal31 opened the pull request:
[1]This commit updates the go-ceph version to use the latest branch
[2] This commit addressed few todo, replacing string comparison with ErrNotExist code of cephFS and rbd

riya-singhal31 committed:
rbd: add check for EncryptionTypeNone

this commit adds the validation for encryption
value as false, and sets the type as none

Signed-off-by: riya-singhal31 <rsinghal@redhat.com>

nixpanic committed:
util: add health-checker for periodic filesystem checks

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
cephfs: enable VolumeCondition with new health-checker

The HealthChecker is configured to use the Staging path pf the volume,
with a `.csi/` subdirectory. In the future this directory could be a
directory that is not under the Published directory.

Fixes: #4219
Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
doc: Add initial design notes for the Health Checker

Signed-off-by: Niels de Vos <ndevos@ibm.com>

nixpanic committed:
util: if FilesystemNodeGetVolumeStats succeeds the volume is healthy

When FilesystemNodeGetVolumeStats() succeeds, the volume must be
healthy. This can be included in the VolumeCondition CSI message by
default.

Checks that detect an abnormal VolumeCondition should prevent calling
FilesystemNodeGetVolumeStats() as it is possible that the function will
hang.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

iPraveenParihar committed:
cephfs: remove snapshot protect/unprotect

This commit eliminates the code for protecting and unprotecting
snapshots, as the functionality to protect and unprotect snapshots
is being deprecated.

Signed-off-by: Praveen M <m.praveen@ibm.com>

riya-singhal31 committed:
Merge branch 'ceph:devel' into devel

iPraveenParihar committed:
cleanup: removed unused SnapshotInfo from CheckSnapExists()

Signed-off-by: Praveen M <m.praveen@ibm.com>

Madhu-1 committed:
doc: add pending release notes

Keeping track of changes between releases
and fetching that information during
release is difficult, Adding a doc to
keep track of the changes between major
releases which helps during release.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
e2e: add multiple labels to node

update the e2e code to add multiple
labels to the node at a time.

fixes: #4146

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
e2e: remove multiple labels to node

update the e2e code to remove multiple
labels to the node at a time.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
util: remove deprecated grpc metrics

This commit removes the deprecated
grpc related code from cephcsi.

fixes: #4122

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
util: rearrange struct to fix lint

re-arrange the struct members to
fix below lint issue

```
struct of size 336 bytes could be of size 328 bytes
```

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: remove unwated go-ceph tags

As we are using latest version of go-ceph
which doesnt require any special tags
for the API's cephcsi is consuming.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
util: remove RadosNamespace from configmap

remove RadosNamespace from the configmap main
section as its already added to the rbd section

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
rbd: remove deprecated rbdImageRequiresEncryption

remove support for deprecated
rbdImageRequiresEncryption case.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

nixpanic committed:
doc: remove use of `XXX` in example descriptions

tickgit.com identifies `XXX` as label for TODO's. There is no need to
list the `digest` hash examples in the TODO list, so  replace the `XXX`
and `YYY` examples with `XYZ` and `ZYX`.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

riya-singhal31 commented:
/test ci/centos/mini-e2e-helm/k8s-1.27

Madhu-1 commented on internal/cephfs/core/metadata.go:
```diff
@@ -152,16 +152,18 @@ func (s *subVolumeClient) UnsetAllMetadata(keys []string) error {
-		if err != nil && !strings.Contains(err.Error(), "No such file or directory") {
-			return fmt.Errorf("failed to unset metadata key %q on subvolume %v: %w", key, s, err)
+		if err != nil {
+			if errors.As(err, &errno) && errno == syscall.ENOENT {
```
@riya-singhal31 please add an E2E to ensure this working as expected or please test this manually by removing the metadata in the backend.

riya-singhal31 replied:
@Madhu-1 I'm replacing this errno with the actual errors that are returned by go-ceph, like for ENOENT they have "ErrNotExist "https://github.com/ceph/go-ceph/blob/master/rbd/errors.go#L77 .
But for cephFS, they have the error as private https://github.com/ceph/go-ceph/blob/master/cephfs/errors.go#L64, so sending a patch there to make it a public and then will modify this PR to direct compare the errors using errors.Is(), that might look more better.

Madhu-1 commented on internal/rbd/rbd_util.go:
```diff
@@ -2141,18 +2142,23 @@ func (rv *rbdVolume) setAllMetadata(parameters map[string]string) error {
-		// TODO: replace string comparison with errno.
-		if err != nil && !strings.Contains(err.Error(), "No such file or directory") {
-			return fmt.Errorf("failed to unset metadata key %q on %q: %w", key, rv, err)
+		if err != nil {
```
please add an E2E to ensure this working as expected or please test this manually by removing the metadata in the backend.

riya-singhal31 committed:
rebase: update github.com/ceph/go-ceph to latest

this commit updates the go-ceph version to use the
latest(master) branch so as to be able to import
cephfs.

Signed-off-by: Riya Singhal <rsinghal@redhat.com>

riya-singhal31 commented:
Updated the PR, just a minor replacement of string check with error code. 
cc: @Madhu-1 , @nixpanic
